RadixWallet/Clients/TokenPriceClient/TokenPriceClient+Live.swift

```diff
@@ -0,0 +1,74 @@
+extension TokenPricesClient {
+	public static let liveValue: TokenPricesClient = {
+		@Dependency(\.httpClient) var httpClient
+		@Dependency(\.jsonDecoder) var jsonDecoder
+		@Dependency(\.jsonEncoder) var jsonEncoder
+		@Dependency(\.cacheClient) var cacheClient
+		let rootURL = URL(string: "https://dev-token-price.extratools.works")!
+
+		@Sendable
+		func getTokenPrices(_ fetchRequest: FetchPricesRequest) async throws -> TokenPrices {
+			let pricesEndpoint = rootURL.appending(component: "price").appending(component: "tokens")
+			var urlRequest = URLRequest(url: pricesEndpoint)
+			urlRequest.httpMethod = "POST"
+			urlRequest.httpBody = try jsonEncoder().encode(fetchRequest)
+
+			urlRequest.allHTTPHeaderFields = [
+				"accept": "application/json",
+				"Content-Type": "application/json",
+			]
+
+			let data = try await httpClient.executeRequest(urlRequest)
+			let decodedResponse = try jsonDecoder().decode(TokensPriceResponse.self, from: data)
+			return .init(decodedResponse)
+		}
+
+		return .init(
+			getTokenPrices: { request in
+				try await cacheClient.withCaching(
+					cacheEntry: .tokenPrices(request.currency),
+					invalidateCached: {
+						Array($0.keys) != request.tokens ? .cachedIsInvalid : .cachedIsValid
+					},
+					request: { try await getTokenPrices(request) }
+				)
+			}
+		)
+	}()
+}
+
+extension TokenPricesClient.TokenPrices {
+	fileprivate init(_ tokenPricesResponse: TokensPriceResponse) {
+		self = tokenPricesResponse.tokens.reduce(into: [:]) { partialResult, next in
+			let roundedToRETPrecision = next.price.roundDoubleToDecimalPlaces(Int(RETDecimal.maxDivisibility - 10))
```

Comment:
hmm why always round to 8 places (18 - 10)? I don't understand where 8 comes from?

Reply:
Ah the `10` is the same as `formatter.maximumFractionDigits = 10` ?? Prefer using a constant for it?